crates/rpc/rpc/src/eth/signer.rs

```diff
@@ -38,13 +39,35 @@ pub(crate) trait EthSigner: Send + Sync {
     fn sign_typed_data(&self, address: Address, payload: &TypedData) -> Result<Signature>;
 }
 
+dyn_clone::clone_trait_object!(EthSigner);
+
 /// Holds developer keys
+#[derive(Clone)]
 pub(crate) struct DevSigner {
     addresses: Vec<Address>,
     accounts: HashMap<Address, SecretKey>,
 }
 
+#[allow(dead_code)]
 impl DevSigner {
+    pub(crate) fn random() -> Box<dyn EthSigner> {
+        let mut signers = Self::random_signers(1);
+        signers.pop().expect("expect to generate at leas one signer")
+    }
+
+    pub(crate) fn random_signers(num: u32) -> Vec<Box<dyn EthSigner + 'static>> {
+        let mut signers = Vec::new();
+        for _ in 0..num {
+            let (pk, sk) = secp256k1::generate_keypair(&mut rand::thread_rng());
+
+            let address = reth_primitives::public_key_to_address(sk);
```

Comment:
I believe the order is flipped here should be (sk, pk)?

Reply:
yes! fixed